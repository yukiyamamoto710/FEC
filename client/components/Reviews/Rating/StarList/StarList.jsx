import React from 'react';
import PropTypes from 'prop-types';
import StarRatingBar from './StarRatingBar/StarRatingBar';

const StarList = (props) => {
  const {
    stars,
    perArray,
    starClicked,
  } = props;

  const handleClickStar = (event) => {
    starClicked(Number(event.target.id));
  };
  return (
    <div
      data-testid="starList"
      className="starListContainer"
    >
      {stars.map((i) => (
        <div key={i}>
          <div
            className="starListBase"
          >
            <button
              data-testid={`startListButton${i}`}
              type="button"
              id={i}
              className="starListText"
              onClick={handleClickStar}
            >
              {`${i} Stars`}
            </button>
            <StarRatingBar per={perArray[i - 1]} />
          </div>
        </div>
      ))}
    </div>
  );
};

StarList.propTypes = {
  stars: PropTypes.arrayOf(PropTypes.string),
  perArray: PropTypes.arrayOf(PropTypes.number),
  starClicked: PropTypes.func,
};

StarList.defaultProps = {
  stars: [],
  perArray: [],
  starClicked: () => 1,
};

export default StarList;
