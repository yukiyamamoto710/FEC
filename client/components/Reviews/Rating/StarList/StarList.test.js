/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
  fireEvent,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import StarList from './StarList';

afterEach(cleanup);

it('should render', () => {
  const { getByTestId } = render(<StarList />);
  expect(getByTestId('starList')).toBeInTheDocument();
});

it('should trigger', () => {
  const testfunc = jest.fn();
  const testArray = ['5', '4', '3', '2', '1'];
  const { getByTestId } = render(<StarList stars={testArray} starClicked={testfunc} />);
  fireEvent.click(getByTestId('startListButton1'));
  expect(testfunc).toHaveBeenCalledTimes(1);
  fireEvent.click(getByTestId('startListButton2'));
  expect(testfunc).toHaveBeenCalledTimes(2);
});

it('should have default starClicked()', () => {
  expect(StarList.defaultProps.starClicked()).toBeDefined();
});

it('matches snapShot', () => {
  const tree = renderer.create(<StarList />).toJSON();
  expect(tree).toMatchSnapshot();
});
