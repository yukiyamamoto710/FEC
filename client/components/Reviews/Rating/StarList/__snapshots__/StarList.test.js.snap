// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="starListContainer"
  data-testid="starList"
/>
`;
